packages/manager/apps/catalog-revamp/e2e/FilterAndSearch.e2e.ts

```diff
@@ -0,0 +1,100 @@
+import { test, expect, Page, ElementHandle } from '@playwright/test';
+import '@playwright-helpers/login';
+import * as translation from '../src/public/translations/catalog-revamp/Messages_fr_FR.json';
+
+const clickButtonByIndex = async (page: Page, index: number): Promise<void> => {
+  const buttons = await page.$$('osds-button');
+  const button = buttons[index];
+  if (button) await button.click();
+};
+
+const clickCheckboxByIndex = async (
+  page: Page,
+  index: number,
+): Promise<void> => {
+  const checkboxes = await page.$$('osds-checkbox-button');
+  const checkbox = checkboxes[index];
+  if (checkbox) await checkbox.click();
+};
+
+const waitForProducts = async (
+  page: Page,
+  expectedProductCount: number,
+): Promise<ElementHandle[]> => {
+  await page.waitForFunction(
+    ({ selector, expectedCount }) =>
+      document.querySelectorAll(selector).length === expectedCount,
+    { selector: 'msc-tile', expectedCount: expectedProductCount },
+  );
+  return page.$$('msc-tile');
+};
+
+const validateProductCategories = async (
+  products: ElementHandle[],
+  expectedCategories: string[],
+): Promise<void> => {
+  const checks = products.map(async (product, index) => {
+    const tileTypeElement = await product.$('.tile-type');
+    expect(tileTypeElement).toBeTruthy();
+
+    const tileTypeText = await tileTypeElement?.textContent();
+    expect(tileTypeText?.trim()).toBe(expectedCategories[index]);
+
+    expect(await product.$('.tile-title')).toBeTruthy();
+  });
+
+  await Promise.all(checks);
+};
+
+test('should filter results based on Bare Metal Cloud universe', async ({
+  page,
+}) => {
+  await clickButtonByIndex(page, 0);
```

Comment:
we can add 'name'  attribute on button or checkbox to be more specific for the test ?

Reply:
Done